File: app/src/main/java/com/datarockets/mnchkn/fragments/players/di/PlayerModule.java
package com.datarockets.mnchkn.fragments.players.di;

import com.datarockets.mnchkn.di.ActivityScope;
import com.datarockets.mnchkn.fragments.players.PlayerInteractor;
import com.datarockets.mnchkn.fragments.players.PlayerPresenter;
import com.datarockets.mnchkn.fragments.players.PlayerPresenterImpl;
import com.datarockets.mnchkn.fragments.players.PlayerView;

import dagger.Module;
import dagger.Provides;

@Module
public class PlayerModule {

    private PlayerView mPlayerView;

    public PlayerModule(PlayerView playerView) {
        this.mPlayerView = playerView;
    }

    @ActivityScope
    @Provides
    public PlayerView providesPlayerView() {
        return mPlayerView;
    }

    @ActivityScope
    @Provides
    public PlayerPresenter providesPlayerPresenter(PlayerView playerView,
                                                   PlayerInteractor playerInteractor) {
        return new PlayerPresenterImpl(playerView, playerInteractor);
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/players/di/PlayerModule.kt
package com.datarockets.mnchkn.fragments.players.di

import com.datarockets.mnchkn.di.ActivityScope
import com.datarockets.mnchkn.fragments.players.PlayerInteractor
import com.datarockets.mnchkn.fragments.players.PlayerPresenter
import com.datarockets.mnchkn.fragments.players.PlayerPresenterImpl
import com.datarockets.mnchkn.fragments.players.PlayerView

import dagger.Module
import dagger.Provides

@Module class PlayerModule(private val mPlayerView: PlayerView) {

    @ActivityScope @Provides
    fun providesPlayerView(): PlayerView {
        return mPlayerView
    }

    @ActivityScope @Provides
    fun providesPlayerPresenter(playerView: PlayerView,
                                playerInteractor: PlayerInteractor): PlayerPresenter {
        return PlayerPresenterImpl(playerView, playerInteractor)
    }

}
